File: mobile/lib/features/nutrition/domain/ingredient.dart
// =====================================================================
// KineticOs — Catálogo de ingredientes (mirror de NutritionDtos en backend).
// =====================================================================
class Ingredient {
  final int id;
  final String name;
  final String? category;
  final String baseUnit;
  final double caloriesPer100g;
  final double proteinPer100g;
  final double carbsPer100g;
  final double fatPer100g;
  final double fiberPer100g;
  final bool aiGenerated;

  const Ingredient({
    required this.id,
    required this.name,
    this.category,
    required this.baseUnit,
    required this.caloriesPer100g,
    required this.proteinPer100g,
    required this.carbsPer100g,
    required this.fatPer100g,
    required this.fiberPer100g,
    this.aiGenerated = false,
  });

  factory Ingredient.fromJson(Map<String, dynamic> json) => Ingredient(
        id: json['id'] as int,
        name: json['name'] as String,
        category: json['category'] as String?,
        baseUnit: json['baseUnit'] as String,
        caloriesPer100g: (json['caloriesPer100g'] as num).toDouble(),
        proteinPer100g: (json['proteinPer100g'] as num).toDouble(),
        carbsPer100g: (json['carbsPer100g'] as num).toDouble(),
        fatPer100g: (json['fatPer100g'] as num).toDouble(),
        fiberPer100g: (json['fiberPer100g'] as num).toDouble(),
        aiGenerated: json['aiGenerated'] as bool? ?? false,
      );
}

File: backend/modules/nutrition/src/main/java/com/micoach/nutrition/domain/Ingredient.java
package com.micoach.nutrition.domain;

import lombok.Getter;

import java.math.BigDecimal;
import java.time.Instant;

/**
 * Catálogo de alimentos con macros por 100 g (tabla nutrition_ingredients).
 */
@Getter
public class Ingredient {

    private final Long id;
    private final String name;
    private final String category;
    private final String baseUnit;
    private final BigDecimal caloriesPer100g;
    private final BigDecimal proteinPer100g;
    private final BigDecimal carbsPer100g;
    private final BigDecimal fatPer100g;
    private final BigDecimal fiberPer100g;
    private final boolean aiGenerated;
    private final boolean active;
    private final Instant createdAt;
    private final Instant updatedAt;

    private Ingredient(Long id, String name, String category, String baseUnit,
                       BigDecimal caloriesPer100g, BigDecimal proteinPer100g, BigDecimal carbsPer100g,
                       BigDecimal fatPer100g, BigDecimal fiberPer100g, boolean aiGenerated, boolean active,
                       Instant createdAt, Instant updatedAt) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.baseUnit = baseUnit;
        this.caloriesPer100g = caloriesPer100g;
        this.proteinPer100g = proteinPer100g;
        this.carbsPer100g = carbsPer100g;
        this.fatPer100g = fatPer100g;
        this.fiberPer100g = fiberPer100g;
        this.aiGenerated = aiGenerated;
        this.active = active;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public static Ingredient restore(Long id, String name, String category, String baseUnit,
                                     BigDecimal caloriesPer100g, BigDecimal proteinPer100g,
                                     BigDecimal carbsPer100g, BigDecimal fatPer100g, BigDecimal fiberPer100g,
                                     boolean aiGenerated, boolean active, Instant createdAt, Instant updatedAt) {
        return new Ingredient(id, name, category, baseUnit, caloriesPer100g, proteinPer100g, carbsPer100g,
                fatPer100g, fiberPer100g, aiGenerated, active, createdAt, updatedAt);
    }
}
